[CI] EsqlFunctionRegistryTests testRegisteredFunctionHaveTests failing

**Build Scans:**
- [elasticsearch-periodic #11985 / release-tests](https://gradle-enterprise.elastic.co/s/tqcykenohtoca)
- [elasticsearch-periodic #11985 / release-tests](https://gradle-enterprise.elastic.co/s/rouh3n6q2mqzi)
- [elasticsearch-pull-request #127601 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/brtxncuzxc53g)
- [elasticsearch-pull-request #127566 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/nig673wc43cau)
- [elasticsearch-pull-request #127351 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/7kxzlg577mqe2)
- [elasticsearch-pull-request #127333 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/b6ak3tbjvafp4)
- [elasticsearch-pull-request #127291 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/fgbpkqcu4ksny)
- [elasticsearch-pull-request #127258 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/6jtn3g5etb3jg)
- [elasticsearch-pull-request #127211 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/vm3jtbn26m3de)
- [elasticsearch-pull-request #127202 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/d5hlmnwfe2l4a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistryTests.testRegisteredFunctionHaveTests" -Dtests.seed=59700DEF7E317431 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=ha-Latn-NG -Dtests.timezone=Canada/Mountain -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistryTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRegisteredFunctionHaveTests'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: function test errorsExpected a list containing
 0: "org.elasticsearch.xpack.esql.expression.function.aggregate.AbsentOverTimeErrorTests is missing"
 1: "org.elasticsearch.xpack.esql.expression.function.aggregate.AvgOverTimeErrorTests is missing"
 2: "org.elasticsearch.xpack.esql.expression.function.aggregate.CountDistinctErrorTests is missing"
 3: "org.elasticsearch.xpack.esql.expression.function.aggregate.CountDistinctOverTimeErrorTests is missing"
 4: "org.elasticsearch.xpack.esql.expression.function.aggregate.CountOverTimeErrorTests is missing"
 5: "org.elasticsearch.xpack.esql.expression.function.aggregate.DeltaErrorTests is missing"
 6: "org.elasticsearch.xpack.esql.expression.function.aggregate.DerivErrorTests is missing"
 7: "org.elasticsearch.xpack.esql.expression.function.aggregate.FirstOverTimeErrorTests is missing"
 8: "org.elasticsearch.xpack.esql.expression.function.aggregate.IdeltaErrorTests is missing"
 9: "org.elasticsearch.xpack.esql.expressio
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 14 consecutive failures in step checkpart3_release-tests
- [main] 17 failures in test testRegisteredFunctionHaveTests (1.8% fail rate in 954 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 14 failures in step checkpart3_release-tests (100.0% fail rate in 14 executions)
- [main] 15 failures in pipeline elasticsearch-pull-request (2.4% fail rate in 621 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -399,9 +399,6 @@ tests:
 - class: org.elasticsearch.test.apmintegration.MetricsApmIT
   method: testApmIntegration {withOTel=true}
   issue: https://github.com/elastic/elasticsearch/issues/144072
-- class: org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistryTests
-  method: testRegisteredFunctionHaveTests
-  issue: https://github.com/elastic/elasticsearch/issues/144076
 - class: org.elasticsearch.xpack.inference.integration.AuthorizationTaskExecutorIT
   method: testCreatesChatCompletion_AndThenCreatesTextEmbedding
   issue: https://github.com/elastic/elasticsearch/issues/138012

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistryTests.java
@@ -7,6 +7,7 @@
 
 package org.elasticsearch.xpack.esql.expression.function;
 
+import org.elasticsearch.Build;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.compute.expression.ExpressionEvaluator;
 import org.elasticsearch.test.ESTestCase;
@@ -240,6 +241,10 @@ public class EsqlFunctionRegistryTests extends ESTestCase {
     }
 
     public void testRegisteredFunctionHaveTests() {
+        assumeTrue("""
+            Some functions are missing when outside of snapshot and we really
+            only care about the superset of all functions. Just skip this test
+            on release builds.""", Build.current().isSnapshot());
         EsqlFunctionRegistry registry = new EsqlFunctionRegistry().snapshotRegistry();
         Set<String> errors = new TreeSet<>();
         for (FunctionDefinition def : registry.listFunctions()) {

